flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -445,11 +457,11 @@ public void endInput() throws IOException {
    * avro serialized bytes.
    */
   private byte[] writeToManifest(long checkpointId) throws IOException {
-    if (writeResultsOfCurrentCkpt.isEmpty()) {
+    if (writeResultsSinceLastSnapshot.isEmpty()) {
       return EMPTY_MANIFEST_DATA;
     }
-
-    WriteResult result = WriteResult.builder().addAll(writeResultsOfCurrentCkpt).build();
+    List<WriteResult> writeResults = writeResultsSinceLastSnapshot.get(checkpointId);
```

Comment:
Don't we need to go though all of the content of the `writeResultsSinceLastSnapshot` and create a manifest for all of them?

Reply:
I looped `writeResultsSinceLastSnapshot` before this method and created a manifest for each checkpoint.

Reply:
Ok. So you did the loop in a different place than I thought.
Thx